lib/ts/recipe/emailpassword/components/features/signInAndUp/index.tsx

```diff
@@ -19,228 +19,276 @@
 /** @jsx jsx */
 import { jsx } from "@emotion/react";
 import * as React from "react";
-import { PureComponent, Fragment } from "react";
+import { Fragment } from "react";
 
-import { FormFieldThemeProps } from "../../../types";
+import {
+    EmailPasswordSignInAndUpAction,
+    EmailPasswordSignInAndUpChildProps,
+    FormFieldThemeProps,
+} from "../../../types";
 import SignInAndUpTheme from "../../themes/signInAndUp";
 import { FeatureBaseProps, NormalisedFormField } from "../../../../../types";
-import { getRedirectToPathFromURL } from "../../../../../utils";
+import { getQueryParams, getRedirectToPathFromURL } from "../../../../../utils";
 import FeatureWrapper from "../../../../../components/featureWrapper";
 import { SignInAndUpState, RecipeInterface } from "../../../types";
 import Recipe from "../../../recipe";
 import { ComponentOverrideContext } from "../../../../../components/componentOverride/componentOverrideContext";
 import { defaultTranslationsEmailPassword } from "../../themes/translations";
+import { useMemo } from "react";
+import { useCallback } from "react";
+import { Dispatch } from "react";
 
-type PropType = FeatureBaseProps & {
-    recipe: Recipe;
-};
-
-class SignInAndUp extends PureComponent<PropType, SignInAndUpState> {
-    constructor(props: PropType) {
-        super(props);
-
-        this.state = {
+export const useFeatureReducer = (recipe: Recipe | undefined) => {
+    return React.useReducer(
+        (oldState: SignInAndUpState, action: EmailPasswordSignInAndUpAction) => {
+            switch (action.type) {
+                case "setSignIn":
+                    return {
+                        ...oldState,
+                        isSignUp: false,
+                    };
```

Comment:
is this missing a break;?

Reply:
The return statements return from the entire function, making breaks not necessary.